pkg/auth/webapp/session_authflow.go

```diff
@@ -774,6 +774,19 @@ func (s *AuthflowScreenWithFlowResponse) Advance(p string, result *Result) {
 	result.RedirectURI = u.String()
 }
 
+// RedirectToFinish is a fix for https://linear.app/authgear/issue/DEV-1793/investigate-sign-in-directly-with-httpsaccountsportalauthgearcom-crash
+// We need Turbo to visit /finish with a full browser redirect,
+// so CSP and connect-src will not kick in.
+func (s *AuthflowScreenWithFlowResponse) RedirectToFinish(route string, result *Result) {
+	q := url.Values{}
+	q.Set(AuthflowQueryKey, s.Screen.StateToken.XStep)
+	u, _ := url.Parse(route)
+	u.RawQuery = q.Encode()
+
+	result.NavigationAction = "redirect"
+	result.RedirectURI = u.String()
+}
```

Comment:
Sorry I still don't quite understand the cause. Why using `advance` will cause the problem but `redirect` will not?

Reply:
If it is `advance`, then Turbo will visit `/finish` with XHR. The response of the XHR is 302 with a Location. According to MDN, CSP `connect-src` affects XHR. If Location is not allowed by `connect-src`, the redirect is blocked.

Reply:
If it is `redirect`, then our script is programmed to visit the url with `window.location.href = THE_URL`. And typical HTTP redirect is not affected by CSP.

Reply:
I understand the issue after I read the notes you've written inside the issue.
So it is not `/authflow/v2/finish` itself have problem, but the redirect_uri which finally being redirected to was blocked.

Reply:
Yes.